src/components/SearchReps.vue

```diff
@@ -196,9 +196,29 @@ export default {
         },
         async CreateRepList() {
             try {
-                const res = await axios.get(
+
+                // check postal code is valid with regex
+                let res = ''
+                let isPostalCodeValid =  /(^\d{5}$)|(^\d{5}-\d{4}$)/.test(this.postalCode);
+                if(!isPostalCodeValid) {
+                  res = await axios.get(
                     '/api/representatives/' + this.postalCode
                 )
+
+                }else{
+                  // check if street address is valid with a flexible regex. This validation is not perfect, but caches common cases
+                  // [a-zA-Z0-9\s]	Any single character in the range a-z or A-Z or 0-9 or whitespace
+                  // '-	Matches the characters - (case sensitive)
+                  // * 0 or more times
+                  // $  end of string
+                  let streetAddressValid = /^[a-zA-Z0-9\s,'-]*$/.test(this.postalCode);
```

Comment:
The name of the parameter should change from `postalCode` to something like `searchTerm`

When searching by address, the [API request body](https://cicero.azavea.com/docs/#address) is split up into multiple parts like `search_address`, `search_city`, etc.  There needs to be more complex parsing to split up the search term, or we need a new way to provide multiple separate inputs

Reply:
@thyeggman 
But the search_address parameter is enough to generate a successful and logical response that returns several districts. I have tried https://cicero.azavea.com/v3.1/legislative_district?search_loc=990+spring+garden+st+philadelphia+pa&format=json&key=405f15046940ce2a712b41c0f442b1ac2e57e3b9
. I feel that if the team wants it to include more parameters like search_city and even more than that, the input cannot be a simple input text, it should be stricter than this. Regex enough cannot or should not handle so many edge cases, I believe.

Reply:
Ah, you're right, my bad!  For some reason my brain completely ignored the alternate `search_loc` parameter that can be used 😅

Reply:
No worries. I will update the rest of the code to fix the issues you found.